cores/esp8266/Esp.cpp

```diff
@@ -698,15 +698,15 @@ static SpiFlashOpResult spi_flash_write_puya(uint32_t offset, uint32_t *data, si
             bytesLeft = 0;
         }
         size_t bytesAligned = (bytesNow + 3) & ~3;
-        rc = spi_flash_read(pos, flash_write_puya_buf, bytesAligned);
+        rc = spi_flash_read(pos, flash_write_puya_buf, bytesNow);
```

Comment:
The API docs seem to specify that the size needs to be quad-aligned, too ( https://www.espressif.com/sites/default/files/documentation/2c-esp8266_non_os_sdk_api_reference_en.pdf )

![image](https://user-images.githubusercontent.com/11875/89748111-76a8ee00-da76-11ea-9a05-35dad5db1964.png)

Sorry for the image paste.  The PDF wouldn't text-copy properly for me.